lib/trino-parquet/src/main/java/io/trino/parquet/reader/PageReader.java

```diff
@@ -31,24 +30,21 @@
 {
     private final CompressionCodecName codec;
     private final long valueCount;
-    private final List<DataPage> compressedPages;
+    private final LinkedList<DataPage> compressedPages;
     private final DictionaryPage compressedDictionaryPage;
     private final OffsetIndex offsetIndex;
     private int pageIndex;
 
     public PageReader(CompressionCodecName codec,
-                      List<DataPage> compressedPages,
+                      LinkedList<DataPage> compressedPages,
```

Comment:
This needs some documentation. This class now holds on to the list that's passed in and mutates it, so the ownership semantics need to be clear to the caller.

Reply:
Where would you suggest to add that? In-line comments on the creation site, on this constructor, renaming the variable to reflect the semantics, or all of the above?

Reply:
I would add it to the Javadocs for the constructor

Reply:
Added